### PR TITLE
Command line script to query enumerateDatasets endpoint data

### PR DESCRIPTION
**Pull request by eparkmonster**
## Why
We want to be able to pull dataset information like ID when querying Jades' enumerateDatasets endpoint given a provided name

[Relevant ticket](https://broadinstitute.atlassian.net/browse/DSPDC-1728)

## This PR
Adds a function "enumerateDataset" to grab all dataset information that comes from querying based on a given name.
Also adds command line argument "-q"  for querying that dataset information.

### CODE REVIEW

File: orchestration/dagster_orchestration/hca_manage/dataset.py [hunk 1]
```diff
@@ -51,3 +54,9 @@ def create_dataset(args: argparse.Namespace, host: str) -> JobId:
         billing_profile_id=args.billing_profile_id,
         schema_path=args.schema_path
     )
+
+
+def query_dataset(args: argparse.Namespace, host: str) -> str:
+    hca = HcaManage(environment=args.env, data_repo_client=get_api_client(host=host), dataset=args.dataset_name)
+    print(hca.enumerate_dataset())
+    return hca.enumerate_dataset()
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/dataset.py:62]: 🔧 This function should not have a return and have a return type annotation  `None`; it's redundant to return the dataset payload as nothing is consuming it.

File: orchestration/dagster_orchestration/hca_manage/manage.py [hunk 2]
```diff
@@ -240,6 +240,10 @@ def get_dataset_id(self) -> str:
         response = self.data_repo_client.enumerate_datasets(filter=self.dataset)
         return response.items[0].id  # type: ignore # data repo client has no type hints, since it's auto-generated
 
+    def enumerate_dataset(self) -> str:
+        response = f"{self.data_repo_client.enumerate_datasets(filter=self.dataset)}"
```
Comment (aherbst-broad) [inline orchestration/dagster_orchestration/hca_manage/manage.py:244]: ❓ What happens if we pass in a bogus dataset name? Will the client throw an exception or just return nothing?

**Review by aherbst-broad — COMMENTED** [review]
Looks good, just one tweak to a return type and an error handling question.

File: orchestration/dagster_orchestration/hca_manage/dataset.py (hunk 1 above)
Comment (eparkmonster) [reply orchestration/dagster_orchestration/hca_manage/dataset.py:62]: ok makes sense, will change

File: orchestration/dagster_orchestration/hca_manage/manage.py (hunk 2 above)
Comment (eparkmonster) [reply orchestration/dagster_orchestration/hca_manage/manage.py:244]: Passing in bogus dataset names will just return an empty list.

**Review by aherbst-broad — APPROVED** [review]
lgtm

**Review by quazi-broad — APPROVED** [review]
looks good to me!

**Pull request merged**